### ISSUE TITLE
ENFILE and EMFILE errors when working with large repositories

### ISSUE THREAD

**Issue by Bobnix** [forcedotcom/cli#366]
When performing a force:source:deploy or force:source:convert to work with a sandbox, we are getting frequent but not consistent ENFILE and EMFILE errors on Mac and Windows respectively.

> { Error: ENFILE: file table overflow, open '/Users/bob.roberts/.local/share/sfdx/node_modules/buffer-crc32/index.js'
    at Object.fs.openSync (fs.js:646:18)
    at Object.fs.readFileSync (fs.js:551:33)
    at Object.Module._extensions..js (module.js:662:20)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Module._load (module.js:497:3)
    at LazyLoader.loadIfNeeded (/Users/bob.roberts/Workspace/Customer-Portal/node_modules/@salesforce/lazy-require/lib/LazyLoader.js:63:19)
    at Object.get (/Users/bob.roberts/Workspace/Customer-Portal/node_modules/@salesforce/lazy-require/lib/LazyLoader.js:219:28)
    at module.exports.ZipArchiveOutputStream._appendBuffer (/Users/bob.roberts/.local/share/sfdx/node_modules/compress-commons/lib/archivers/zip/zip-archive-output-stream.js:71:21)
    at module.exports.ArchiveOutputStream.entry (/Users/bob.roberts/.local/share/sfdx/node_modules/compress-commons/lib/archivers/archive-output-stream.js:86:10)
    at module.exports.ZipStream.entry (/Users/bob.roberts/.local/share/sfdx/node_modules/zip-stream/index.js:157:49)
    at Zip.append (/Users/bob.roberts/.local/share/sfdx/node_modules/archiver/lib/plugins/zip.js:40:15)
    at Archiver._moduleAppend (/Users/bob.roberts/.local/share/sfdx/node_modules/archiver/lib/core.js:166:16)
    at Archiver._onQueueTask (/Users/bob.roberts/.local/share/sfdx/node_modules/archiver/lib/core.js:356:8)
    at /Users/bob.roberts/.local/share/sfdx/node_modules/async/dist/async.js:4096:9
    at Object.process (/Users/bob.roberts/.local/share/sfdx/node_modules/async/dist/async.js:2344:17)
  errno: -23,
  code: 'ENFILE',
  syscall: 'open',
  path: '/Users/bob.roberts/.local/share/sfdx/node_modules/buffer-crc32/index.js' }

From some internal testing, it looks like excluding buffer-crc32 from LazyLoad eliminates the error. I see that there is an exclusions parameter but it is not exposed past LazyRequire.

**Comment by clairebianchi** [comment]
@Bobnix  is this still an issue?

**Comment by Bobnix** [comment]
To be fair, I wrote a patch for LazyRequire that my build script picks up and haven't had the problem since. It works for me but I haven't opened a pull request since I don't know if this is too specific of a bug.


```
-        return new LazyRequire(typeCache);
+        return new LazyRequire(typeCache, ['buffer-crc32']);
```

**Comment by MGarfOppLoans** [comment]
this is still an issue @clairebianchi, we are seeing this on our end

**Comment by vc4u** [comment]
I'm getting EMFILE errors after latest update (i did not receive errors before). My error is coming when executing `force:mdapi:convert` command. 

The error I'm getting is:
```
(node:48120) UnhandledPromiseRejectionWarning: Error: EMFILE: too many open files, open '<Path>\force-app\main\default\staticresources\jQuery_UI\jquery-ui-1.9.2.custom\css\ui-darkness\images\ui-bg_inset-soft_30_f58400_1x100.png'
(node:48120) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:48120) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

The static resource in context is a zipped file, with file size of `1,698,512 bytes`. This static resource hasn't been changed in a while in my application and the same script was working fine before the update. And the complete code repository is large approx 35mb in size.

**Comment by amphro** [comment]
I realize this was filed first, but I think we addressed this problem in #494. Please file a new issue with new repo steps if this is still an issue.
